homeassistant/components/device_tracker/__init__.py

```diff
@@ -445,28 +467,33 @@ def load_config(path: str, hass: HomeAssistantType, consider_home: timedelta):
         return []
 
 
-def setup_scanner_platform(hass: HomeAssistantType, config: ConfigType,
+@asyncio.coroutine
+def async_setup_scanner_platform(hass: HomeAssistantType, config: ConfigType,
                            scanner: Any, see_device: Callable):
     """Helper method to connect scanner-based platform to device tracker."""
     interval = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
 
     # Initial scan of each mac we also tell about host name for config
     seen = set()  # type: Any
 
-    def device_tracker_scan(now: dt_util.dt.datetime):
+    @asyncio.coroutine
+    def async_device_tracker_scan(now: dt_util.dt.datetime):
         """Called when interval matches."""
-        for mac in scanner.scan_devices():
+        found_devices = yield from hass.loop.run_in_executor(
+            None, scanner.scan_devices)
+
+        for mac in found_devices:
             if mac in seen:
                 host_name = None
             else:
                 host_name = scanner.get_device_name(mac)
```

Comment:
There is no guarantee that `get_device_name` does not do I/O. We probably want to run 1 job in the executor that calls `scanner.scan_devices` and will also fetch the hosts for all devices that have not been seen yet.